src/zkevm_specs/util/arithmetic.py

```diff
@@ -15,10 +15,13 @@ def linear_combine(seq: Sequence[Union[int, FQ]], base: FQ, range_check: bool =
     >>> assert linear_combine([1, 2, 3], r) == 1 + 2 * r + 3 * r**2
     """
     result = FQ.zero()
+    base_int = base.n if isinstance(base, FQ) else base
+    if base_int == 0:
+        return result
     for limb in reversed(seq):
         if range_check:
             limb_int = limb.n if isinstance(limb, FQ) else limb
-            assert 0 <= limb_int < 256, "Each byte should fit in 8-bit"
+            assert 0 <= limb_int < base_int, "Each limb should fit in base"
```

Comment:
That is partially OK for me. As it keeps the previous functionality while making the gadget more flexible. But I'm not sure if this will significantly impact the current circuit implementations.

I'm not specifically against this, but would love to hear from others what opinions they have.

Reply:
@CPerezz is it possible to have your email for other zkevm related discussions?

Reply:
The byte range check should be the expected check here, since this function is only used to linearly combine bytes. So perhaps we could rename this function to a better name to avoid this kind of confusion.

Reply:
Hmm we should maybe then keep this as it was before?? As since @han0110 correctly said, in linear combinations we only use bytes as inputs. Therefore, it doesn't give any advantages to be generic in the assertion.And it's at the same time introducing the issue with the `return` in the `0` case.

Considering all this, might be better to leave the code as it was. And as Han said change the fn name maybe for one like `linear_combine_bytes`.

Reply:
Done.